client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -0,0 +1,271 @@
+    private final Set<IndicesPrivileges> indicesPrivileges;
+    private final Set<ApplicationResourcePrivileges> applicationResourcePrivileges;
+    private final Set<String> runAs;
+    private final Map<String, Object> metadata;
+
+    private Role(Collection<ClusterPrivilege> clusterPrivileges,
+            @Nullable ManageApplicationsPrivilege manageApplicationPrivileges, Collection<IndicesPrivileges> indicesPrivileges,
+            Collection<ApplicationResourcePrivileges> applicationResourcePrivileges, Collection<String> runAs,
+            Map<String, Object> metadata) {
+        // we do all null checks inside the constructor
+        // no cluster privileges are granted unless otherwise specified
+        this.clusterPrivileges = Collections
+                .unmodifiableSet(clusterPrivileges != null ? new HashSet<>(clusterPrivileges) : Collections.emptySet());
+        this.manageApplicationPrivileges = manageApplicationPrivileges;
+        // no indices privileges are granted unless otherwise specified
+        this.indicesPrivileges = Collections
+                .unmodifiableSet(indicesPrivileges != null ? new HashSet<>(indicesPrivileges) : Collections.emptySet());
+        // no application resource privileges are granted unless otherwise specified
+        this.applicationResourcePrivileges = Collections.unmodifiableSet(
+                applicationResourcePrivileges != null ? new HashSet<>(applicationResourcePrivileges) : Collections.emptySet());
+        // no run as privileges are granted unless otherwise specified
+        this.runAs = Collections.unmodifiableSet(runAs != null ? new HashSet<>(runAs) : Collections.emptySet());
+        this.metadata = metadata != null ? Collections.unmodifiableMap(metadata) : Collections.emptyMap();
+    }
+
+    public Set<ClusterPrivilege> getClusterPrivileges() {
+        return clusterPrivileges;
+    }
+
+    public ManageApplicationsPrivilege getManageApplicationPrivileges() {
+        return manageApplicationPrivileges;
+    }
+
+    public Set<IndicesPrivileges> getIndicesPrivileges() {
+        return indicesPrivileges;
+    }
+
+    public Set<ApplicationResourcePrivileges> getApplicationResourcePrivileges() {
+        return applicationResourcePrivileges;
+    }
+
+    public Set<String> getRunAs() {
+        return runAs;
+    }
+
+    public Map<String, Object> getMetadata() {
+        return metadata;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Role that = (Role) o;
+        return clusterPrivileges.equals(that.clusterPrivileges)
+                && Objects.equals(manageApplicationPrivileges, that.manageApplicationPrivileges)
+                && indicesPrivileges.equals(that.indicesPrivileges)
+                && applicationResourcePrivileges.equals(that.applicationResourcePrivileges) && runAs.equals(that.runAs)
```

Comment:
I prefer to see each of these on a line by itself.

Reply:
You're right. will do.